elements/lisk-p2p/test/integration/custom_node_info.ts

```diff
@@ -14,73 +14,28 @@
  */
 import { expect } from 'chai';
 import { P2P } from '../../src/index';
-import { wait } from '../utils/helpers';
-import { platform } from 'os';
+import { createNetwork, destroyNetwork } from '../utils/network_setup';
 
 describe('Custom nodeInfo', () => {
 	let p2pNodeList: ReadonlyArray<P2P> = [];
-	const NETWORK_START_PORT = 5000;
-	const NETWORK_PEER_COUNT = 15;
-	const POPULATOR_INTERVAL = 50;
 
 	before(async () => {
 		sandbox.restore();
 	});
 
 	beforeEach(async () => {
-		p2pNodeList = [...Array(NETWORK_PEER_COUNT).keys()].map(index => {
-			// Each node will have the previous node in the sequence as a seed peer except the first node.
-			const seedPeers =
-				index === 0
-					? []
-					: [
-							{
-								ipAddress: '127.0.0.1',
-								wsPort: NETWORK_START_PORT + index - 1,
-							},
-					  ];
-
-			const nodePort = NETWORK_START_PORT + index;
-
-			return new P2P({
-				seedPeers,
-				wsEngine: 'ws',
-				// A short connectTimeout and ackTimeout will make the node to give up on discovery quicker for our test.
-				connectTimeout: 100,
-				ackTimeout: 200,
-				populatorInterval: POPULATOR_INTERVAL,
-				maxOutboundConnections: 5,
-				maxInboundConnections: 5,
-				nodeInfo: {
-					wsPort: nodePort,
-					nethash:
-						'da3ed6a45429278bac2666961289ca17ad86595d33b31037615d4b8e8f158bba',
-					version: '1.0.1',
-					protocolVersion: '1.1',
-					minVersion: '1.0.0',
-					os: platform(),
-					height: 0,
-					broadhash:
-						'2768b267ae621a9ed3b3034e2e8a1bed40895c621bbb1bbd613d92b9d24e54b5',
-					nonce: `O2wTkjqplHII${nodePort}`,
-					modules: {
-						names: ['test', 'crypto'],
-						active: true,
-					},
-				},
-			});
+		const customModules = () => ({
+			modules: {
+				names: ['test', 'crypto'],
+				active: true,
+			},
 		});
-		await Promise.all(p2pNodeList.map(async p2p => await p2p.start()));
-		await wait(1000);
+
+		p2pNodeList = await createNetwork({ customNodeInfo: customModules });
```

Comment:
This is a good example where the `createNetwork` utility makes the test easier to follow because there aren't many custom properties.